ams_app/views.py

```diff
@@ -0,0 +1,444 @@
+from django.contrib.auth.models import User
+from django.contrib import messages
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.decorators import login_required
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from django.contrib.postgres.search import SearchVector, SearchQuery
+
+from ams_app import forms
+from ams_app.models import Assignment, Submission, Profile
+
+
+def sign_up(request):
+    user_form = forms.UserForm
+    profile_form = forms.ProfileForm
+    if request.method == "POST":
+        user_form = user_form(request.POST)
+        profile_form = profile_form(request.POST)
+        # check whether it's valid:
+        if user_form.is_valid() and profile_form.is_valid():
+            user = User.objects.create_user(
+                username=user_form.cleaned_data["username"],
+                email=user_form.cleaned_data["email"],
+                password=user_form.cleaned_data["password"])
+            user.profile.role = profile_form.cleaned_data["role"]
+            user.profile.matric_number= profile_form.cleaned_data["matric_number"]
+            user.save()
+            messages.success(request, 'User was successfully created.')
+            return redirect('/')
+        for error in user_form.errors.values() or  error in profile_form.errors.values():
+            messages.error(request, error)
+    login_form = forms.LoginForm()
+    context = {
+        "user_form": user_form(),
+        "profile_form": profile_form(),
+        "login": login_form
+    }
+    return render(request, "ams_app/auth.html", context=context)
+
+
+def login_user(request):
+    if request.method == "POST":
+        form = forms.LoginForm(request.POST)
+        if form.is_valid():
+            username = form.cleaned_data["username"]
+            password = form.cleaned_data["password"]
+            user = authenticate(username=username, password=password)
+            if user is not None:
+                login(request, user)
+                messages.success(request, 'User was successfully logged in.')
+                return redirect('/dashboard')
+            else:
+                messages.error(request, 'Username or password is incorrect')
+        else:
+            for error in form.errors.values():
+                messages.error(request, error)
+            return redirect('/')
+
+    login_form = forms.LoginForm()
+    context = {
+        "login": login_form
+    }
+    return render(request, "ams_app/auth.html", context=context)
+
+
+@login_required
+def logout_user(request):
+    logout(request)
+    messages.success(request, 'User was successfully logged out.')
+    return redirect('/')
+
+
+@login_required
+def dashboard(request):
+    assignment_form = forms.AssignmentForm()
+    search_form = forms.AssignmentSearchForm()
+    id = request.user.id
+    user = Profile.objects.get(user_id=id)
+    assignments= Assignment.objects.filter(user_id=id)
```

Comment:
Since `Assignment` has a foreign key relationship with `User`, this could be `request.user.assignment_set.all()` or `request.user.assignments` if you added the `related_name=assignments` in the FK definition in the `Assignment` model